### ISSUE TITLE
iPad Pro. Unable to Checkout. Using Different Billing and Shipping Addresses. 

### ISSUE THREAD

**Issue by giancorderoortiz** [SAP/cloud-commerce-spartacus-storefront#3172]
iPad Pro
iOS 11.0.3

**Steps to Reproduce:** 
1. Open Spartacus storefront in ccv2 env. 
2. Login as registered user.
3. Add products to cart and proceed to checkout. 
4. Add new shipping address or select stored address. 
5. In payment method tab, enter a new credit card, and a new billing address. (De-select "use same shipping address)
6. Enter a new billing address. 
7. Proceed to place order. 

**Expected Results:** 
7. Order placed successfully. 

**Actual Results**: 
Error message displayed. Missing state/province field. Billing address not accepted. 
Unable to add alternative billing address (i.e. different than shipping address).
Unable to complete order even after selecting an OK credit card.
Basically, I have to log out and log back in to start checkout process again.

![billingaddressissue1](https://user-images.githubusercontent.com/46171897/59451656-f2a79780-8dd9-11e9-8721-d3a932546fa4.png)
![billingaddressissue2](https://user-images.githubusercontent.com/46171897/59451658-f3402e00-8dd9-11e9-80b0-b9525a07ae34.png)
![billingaddressissue3](https://user-images.githubusercontent.com/46171897/59451659-f3402e00-8dd9-11e9-8cd9-37c13d9c34c0.png)
![billingaddressissue4](https://user-images.githubusercontent.com/46171897/59451660-f3402e00-8dd9-11e9-93ef-b3fafa44df16.png)

**Comment by kacperknapik** [comment]
@giancorderoortiz 	please provide address that was used in 'new billing address' form. I can't reproduce that error on Mac atm. 
on the second hand I might know what could cause this problem, will check that

**Comment by giancorderoortiz** [comment]
Able to reproduce on Mac, Chrome. Public Spartacus server. 
Billing address used: 
Country: Canada
First name: Yogi
Last name: Bera
Address 1: 444 North Avenue
Address 2: Apt 5
City: Toronto
Zip Code: HTS J2R
Province/State (Field not available)

**Comment by giancorderoortiz** [comment]
![Screen Shot 2019-06-17 at 10 45 15 AM](https://user-images.githubusercontent.com/46171897/59614831-84234c00-90ef-11e9-885e-e68aa26c07eb.png)

**Comment by giancorderoortiz** [comment]
![Screen Shot 2019-06-17 at 10 45 55 AM](https://user-images.githubusercontent.com/46171897/59615078-eda35a80-90ef-11e9-9aaf-e35993302e40.png)
![Screen Shot 2019-06-17 at 10 46 25 AM](https://user-images.githubusercontent.com/46171897/59615087-f005b480-90ef-11e9-9fed-1a78f7e3e425.png)

**Comment by kacperknapik** [comment]
https://github.com/SAP/cloud-commerce-spartacus-storefront/issues/93#issuecomment-460266509

I have to check if the change from the link above was made.

**Comment by WeizhengSap** [comment]
Find another bug, reopen this ticket

**Comment by WeizhengSap** [comment]
Will fix it in another ticket.

**Comment by LTiger14** [comment]
Will be fixed in #3255

**Comment by giancorderoortiz** [comment]
Test successful in dev-com17. 
Able to checkout using a new billing address. 🆗 
Permutations tested:
1. New user -> no stored addresses, nor payment types. 
2. Registered user -> no stored addresses. 
3. Registered user -> stored addresses, stored payment types. 
4. Registered user -> stored addresses.
